packages/appkit/src/adapters/ChainAdapterBlueprint.ts

```diff
@@ -139,12 +139,17 @@ export abstract class AdapterBlueprint<
         w3mThemeVariables: getW3mThemeVariables(themeVariables, themeMode)
       })
     }
-    this.availableConnectors = [
-      ...this.availableConnectors.filter(
-        existing => !connectors.some(newConnector => newConnector.id === existing.id)
-      ),
-      ...connectors
-    ]
+
+    const connectorsAdded = new Set<string>()
+    this.availableConnectors = [...connectors, ...this.availableConnectors].filter(connector => {
+      if (connectorsAdded.has(connector.id)) {
+        return false
+      }
+
+      connectorsAdded.add(connector.id)
+
+      return true
+    })
```

Comment:
Can't you directly do `this.availableConnectors = Array.from(new Set([...connectors, ...this.availableConnectors]))`?

Reply:
for this to work I guess the connectors must have the same object memory reference, if they are different objects with same id it may fail